autocertkit/network_tests.py

```diff
@@ -1002,6 +1002,88 @@ def test_ping(self, session):
         return self._run_test(session)
 
 
+class MulticastTest(IperfTest):
+    """ Subclass that adds multicast specific operations"""
+    MULTICAST_IP = '226.94.1.1'
+    iperf_report_path = '/tmp/iperf.log'
+
+    def run_mulitcast_server(self):
+        """Start the iPerf server listening on a VM"""
+        log.debug('Starting multicast server')
+        self.server_ip = self.get_server_ip('eth0')
+        cmd_str = 'nohup iperf -y csv -s -u -B %s </dev/null &>%s &' \
+            % (self.MULTICAST_IP, self.iperf_report_path)
+        ssh_command(self.server_ip, self.username, self.password,
+                    cmd_str)
+
+    def run_multicast_client(self):
+        """Run test iperf command on droid VM"""
+        log.debug('Starting IPerf client')
+        cmd_str = 'iperf -c %s -u' % self.MULTICAST_IP
+        ssh_command(self.get_client_ip(), self.username, self.password,
+                    cmd_str)
+
+    def validate_multicast(self):
+        cmd_str = 'cat %s' % self.iperf_report_path
+        cmd_result = ssh_command(self.server_ip, self.username,
+                                 self.password, cmd_str)
+        log.debug('Iperf log on multicast server: %s' % cmd_result)
+        if self.MULTICAST_IP not in cmd_result:
+            raise TestCaseError('Error: Multicast test failed. iperf data: %s'
+                                 % cmd_result)
+
+        self.iperf_data = self.parse_iperf_line(cmd_result)
+
+    def run(self):
+        """This classes run test function"""
+        self.deploy_iperf()
+        self.run_mulitcast_server()
+        log.debug('IPerf deployed and multicast server started')
+        self.run_multicast_client()
+
+        iperf_data = self.validate_multicast()
+        return iperf_data
+
+
+class MulticastTestClass(IperfTestClass):
+    """ Subclass that runs multicast test"""
+
+    caps = [REQ_CAP_MULTICAST]
```

Comment:
@bpwook , @rdobson ,
It does look overkill as we never needed to add capabilities (as almost every test was a REQ (required one)), May I should keep it as an action item to add more specific Capabilities to each test and have several capabilities populated. 
For eg JUMBO_FRAMES or CRASHDUMP.

Till date we have been doing that by looking into failed test list but we cannot expect customers to have understanding of that. 
Adding more capabilities does look a good option to me, what say? 

@fillzero 
REQ_CAP_MULTICAST -> CAP_MULTICAST